core/templates/App.ts

```diff
@@ -205,11 +205,20 @@ angular.module('oppia').config([
               return $q(function(resolve, reject) {
                 // Get CSRF token before sending the request.
                 CsrfTokenService.getTokenAsync().then(function(token) {
-                  config.data = $.param({
-                    csrf_token: token,
-                    payload: JSON.stringify(config.data),
-                    source: document.URL
-                  }, true);
+                  // If the request data is of the form Formdata we don't
```

Comment:
is of the type Formdata....

Reply:
I am a still confused regarding the why part here -- i.e. why we don't to add the payload again, maybe an example will help.

Reply:
@ankita240796, I've made the code here a bit more general, let me know if things are still not clear